lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart

```diff
@@ -0,0 +1,228 @@
+class RequiredFluxProps extends RecursiveAstVisitor with ClassSuggestor {
+  ResolvedUnitResult? _result;
+
+  static const fluxPropsMixinName = 'FluxUiPropsMixin';
+
+  @override
+  visitCascadeExpression(CascadeExpression node) {
+    var writesToFluxUiProps = false;
+    var actionsAssigned = false;
+    var storeAssigned = false;
+
+    DartType? fluxActionsType;
+    DartType? fluxStoreType;
+    final cascadeWriteEl = node.staticType?.element;
+    if (cascadeWriteEl is ClassElement) {
+      final maybeFluxUiPropsMixin = cascadeWriteEl.mixins
+          .singleWhereOrNull((e) => e.element.name == fluxPropsMixinName);
+      writesToFluxUiProps = maybeFluxUiPropsMixin != null;
+      fluxActionsType = maybeFluxUiPropsMixin?.typeArguments[0];
+      fluxStoreType = maybeFluxUiPropsMixin?.typeArguments[1];
+    }
+
+    final cascadingAssignments =
+        node.cascadeSections.whereType<AssignmentExpression>();
+    storeAssigned = cascadingAssignments.any((cascade) {
+      final lhs = cascade.leftHandSide;
+      return lhs is PropertyAccess && lhs.propertyName.name == 'store';
+    });
+    actionsAssigned = cascadingAssignments.any((cascade) {
+      final lhs = cascade.leftHandSide;
+      return lhs is PropertyAccess && lhs.propertyName.name == 'actions';
+    });
+
+    if (writesToFluxUiProps && !storeAssigned) {
+      storeAssigned = true;
+      final storeValue =
+          _getNameOfVarOrFieldInScopeWithType(node, fluxStoreType) ?? 'null';
+      yieldNewCascadeSection(node, '..store = $storeValue');
+    }
+
+    if (writesToFluxUiProps && !actionsAssigned) {
+      actionsAssigned = true;
+      final actionsValue =
+          _getNameOfVarOrFieldInScopeWithType(node, fluxActionsType) ?? 'null';
+      yieldNewCascadeSection(node, '..actions = $actionsValue');
+    }
+  }
+
+  void yieldNewCascadeSection(CascadeExpression node, String newSection) {
+    final offset = context.sourceFile.getOffsetOfLineAfter(node.target.offset);
+    yieldPatch(newSection, offset, offset);
+  }
+
+  @override
+  Future<void> generatePatches() async {
+    _result = await context.getResolvedUnit();
+    if (_result == null) {
+      throw Exception(
+          'Could not get resolved result for "${context.relativePath}"');
+    }
+    _result!.unit.accept(this);
+  }
+}
+
+String? _getNameOfVarOrFieldInScopeWithType(AstNode node, DartType? type) {
+  final inScopeVariableDetector = _InScopeVarDetector();
+  // Find top level vars
+  node.thisOrAncestorOfType<CompilationUnit>()?.accept(inScopeVariableDetector);
+  // Find vars declared in top-level fns (like `main()`)
+  node
+      .thisOrAncestorOfType<BlockFunctionBody>()
+      ?.visitChildren(inScopeVariableDetector);
+
+  final inScopeVarName = inScopeVariableDetector.found
+      .firstWhereOrNull((v) {
+        final maybeMatchingType = v.declaredElement?.type;
+        return maybeMatchingType?.element?.name == type?.element?.name;
```

Comment:
Instead of just matching the name, I think we could use `TypeSystem`, which would match any subtypes, and prevent any false positives where in scope variables are different types with the same name.
```dart
  // Usually we'd grab typeSystem from the ResolvedUnitResult, but we don't have access to that
  // in this class, so just get it from the compilation unit.
  final typeSystem = (node.root as CompilationUnitElement).library.typeSystem;
  bool isMatchingType(DartType? maybeMatchingType) =>
      maybeMatchingType != null &&
      typeSystem.isAssignableTo(maybeMatchingType, type);

  final inScopeVarName = inScopeVariableDetector.found
      .firstWhereOrNull((v) => isMatchingType(v.declaredElement?.type))
      ?.declaredElement
      ?.name;
```

Reply:
Addressed in https://github.com/Workiva/over_react_codemod/pull/268/commits/580b82cd2dbf752e89dba8a936ee6a047091274b